### PR TITLE
8311775: [TEST] duplicate verifyHeapDump in several tests

### PR DESCRIPTION
**Pull request by y1yang0**
This is a follow-up patch of #13667. verifyHeapDump is duplicated in several tests, this patch tries to consolidate them into one method.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8311775](https://bugs.openjdk.org/browse/JDK-8311775): [TEST] duplicate verifyHeapDump in several tests (**Enhancement** - P5)

### Reviewers
 * [Kevin Walls](https://openjdk.org/census#kevinw) (@kevinjwalls - Committer) ⚠️ Review applies to [ac581090](https://git.openjdk.org/jdk/pull/15202/files/ac581090a12f6c020525ebe14584799af48b7d4a)
 * [Alex Menkov](https://openjdk.org/census#amenkov) (@alexmenkov - **Reviewer**)
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15202/head:pull/15202` \
`$ git checkout pull/15202`

Update a local copy of the PR: \
`$ git checkout pull/15202` \
`$ git pull https://git.openjdk.org/jdk.git pull/15202/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15202`

View PR using the GUI difftool: \
`$ git pr show -t 15202`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15202.diff">https://git.openjdk.org/jdk/pull/15202.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15202#issuecomment-1670811618)

### CODE REVIEW

**Review by kevinjwalls — APPROVED** [review]
Looks good to me, thanks.
One question on whether we need to change those "throws IOException" to "throws Exception", which is really me thinking if the existing HProfParser parse method really needs to throw Exception?  Maybe it only needs IOException? (snapshot.resolve doesn't throw anything)
But looks good anyway and good to share this code. 8-)

**Comment by plummercj** [comment]
Can you clarify what testing you did? I don't see BasicJMapTest.java being run as part of GHA. I also don't see com/sun/management/HotSpotDiagnosticMXBean/DumpHeap.java being run, which is impacted by your changes.

File: test/hotspot/jtreg/serviceability/sa/TestHeapDumpForInvokeDynamic.java [hunk 1]
```diff
@@ -37,6 +37,7 @@
 import jdk.test.lib.Utils;
 import jdk.test.lib.hprof.parser.HprofReader;
 import jdk.test.lib.hprof.parser.PositionDataInputStream;
+import jdk.test.lib.hprof.HprofParser;
```
Comment (alexmenkov) [inline test/hotspot/jtreg/serviceability/sa/TestHeapDumpForInvokeDynamic.java:40]: PositionDataInputStream and HprofParser imports are not used anymore

**Comment by alexmenkov** [comment]
Need to update copyright year in the files

**Comment by y1yang0** [comment]
Thanks  for reviews!

----------
@kevinjwalls

> One question on whether we need to change those "throws IOException" to "throws Exception", which is really me thinking if the existing HProfParser parse method really needs to throw Exception? Maybe it only needs IOException? (snapshot.resolve doesn't throw anything) But looks good anyway and good to share this code. 8-)

Yes, we must change it to `throws Exception` otherwise we won't compile
```java
/jdk/test/lib/jdk/test/lib/hprof/HprofParser.java:99: error: unreported exception Exception; must be caught or declared to be thrown
            try (Snapshot snapshot = Reader.readFile(dump.getAbsolutePath(), callStack, debugLevel)) {
                          ^
  exception thrown from implicit call to close() on resource variable 'snapshot'
```

----------
@plummercj 

> Can you clarify what testing you did?

All involved tests work in Linux.

File: test/hotspot/jtreg/serviceability/sa/TestHeapDumpForInvokeDynamic.java (hunk 1 above)
Comment (y1yang0) [reply test/hotspot/jtreg/serviceability/sa/TestHeapDumpForInvokeDynamic.java:40]: Thanks for review, all copyright year and unused imports are removed now

File: test/hotspot/jtreg/serviceability/dcmd/gc/HeapDumpTest.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2015, 2017, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2015, 2013, Oracle and/or its affiliates. All rights reserved.
```
Comment (alexmenkov) [inline test/hotspot/jtreg/serviceability/dcmd/gc/HeapDumpTest.java:2]: typo - should be 2023, not 2013

**Comment by plummercj** [comment]
test/lib/jdk/test/lib/hprof/HprofParser.java needs a copyright update.

**Review by alexmenkov — APPROVED** [review]

**Review by plummercj — APPROVED** [review]

**Pull request closed without merging**